src/main/java/com/linecorp/armeria/client/HttpConfigurator.java

```diff
@@ -507,6 +594,51 @@ protected void onCloseRequest(ChannelHandlerContext ctx) throws Exception {
     }
 
     /**
+     * A handler that suppresses the write requests for the HTTP/2 preface string and the initial empty
+     * SETTINGS frame sent by {@link Http2ConnectionHandler}.
+     *
+     * TODO(trustin): Remove this once Netty supports the upgrade via the HTTP/2 preface string properly.
+     *                See: https://github.com/netty/netty/issues/4438
+     */
+    private static final class Http2PrefaceSuppressingHandler extends ChannelOutboundHandlerAdapter {
+
+        private int numWrites;
+
+        @Override
+        public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise) throws Exception {
+            if (!(msg instanceof ByteBuf)) {
+                ctx.write(msg, promise);
+                return;
+            }
+
+            final ByteBuf buf = (ByteBuf) msg;
+
+            switch (buf.readableBytes()) {
```

Comment:
Replaced `Http2PrefaceSuppressingHandler` with `DowngradeHandler` to make it much simpler.